Add deprecation for identifier characters (#786)
neo4j/docs-cypher: modules/ROOT/pages/syntax/naming.adoc

====

[NOTE]
====

> Some more special characters have been deprecated and will require escaping in a Neo4j 6.0, see xref::deprecations-additions-removals-compatibility.adoc#cypher-deprecations-additions-removals-5.15[here] for the comprehensive list of deprecated characters.

Reviewer: ```suggestion
Several special characters have been deprecated and will require escaping in Neo4j 6.0, see xref::deprecations-additions-removals-compatibility.adoc#cypher-deprecations-additions-removals-5.15[here] for the comprehensive list of deprecated characters.
```
